src/contexts/bridge.context.tsx

```diff
@@ -568,14 +588,110 @@ const BridgeProvider: FC<PropsWithChildren> = (props) => {
     [getBridges, refreshBridges]
   );
 
+  const cleanPendingTxs = useCallback(
+    (bridges: Bridge[]): Promise<void> => {
+      if (!env) {
+        return Promise.resolve();
+      }
+
+      const pendingTxs = storage.getPendingTxs(env);
+      const isPendingDepositInApiBridges = (depositTxHash: string) => {
+        return bridges.find((bridge) => {
+          return (
+            (bridge.status === "initiated" || bridge.status === "on-hold") &&
+            bridge.depositTxHash === depositTxHash
+          );
+        });
+      };
+      const isPendingClaimInApiBridges = (claimTxHash: string) => {
+        return bridges.find((bridge) => {
+          return bridge.status === "completed" && bridge.claimTxHash === claimTxHash;
+        });
+      };
+
+      return Promise.all(
+        pendingTxs.map(async (pendingTx) => {
+          if (
+            pendingTx.status === "pending-deposit" &&
+            isPendingDepositInApiBridges(pendingTx.depositTxHash)
+          ) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+          if (
+            pendingTx.status === "pending-claim" &&
+            isPendingClaimInApiBridges(pendingTx.claimTxHash)
+          ) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+
+          const txHash =
+            pendingTx.status === "pending-deposit"
+              ? pendingTx.depositTxHash
+              : pendingTx.claimTxHash;
+          const provider =
+            pendingTx.status === "pending-deposit"
+              ? pendingTx.from.provider
+              : pendingTx.to.provider;
+          const tx = await provider.getTransaction(txHash);
+
+          if (isTxCanceled(tx)) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+
+          if (isTxMined(tx)) {
+            const txReceipt = await provider.getTransactionReceipt(txHash);
+
+            if (hasTxBeenReverted(txReceipt)) {
+              return storage.removePendingTx(pendingTx.depositTxHash);
+            }
+          }
+
+          if (Date.now() > pendingTx.timestamp + PENDING_TX_TIMEOUT) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+        })
+      ).then();
+    },
+    [env]
+  );
+
+  const getPendingBridges = useCallback(
+    async (bridges?: Bridge[]): Promise<PendingBridge[]> => {
+      if (bridges) {
+        await cleanPendingTxs(bridges);
+      }
+
+      if (!env) {
+        throw new Error("Env is not available");
+      }
+
+      return Promise.all(
+        storage.getPendingTxs(env).map(async (tx) => {
+          const chain = env?.chains.find((chain) => chain.networkId === tx.networkId);
+          const fiatAmount =
+            chain && env.fiatExchangeRates.areEnabled
+              ? await getTokenPrice({ token: tx.token, chain })
```

Comment:
I might be missing something here but I think the `fiatAmount` will only match the token price when the token is a stablecoin that matches the selected currency.

In other words, to get the `fiatAmount` of a token (let's say UNI), shouldn't we use `multiplyAmounts`?

Reply:
Buf, good one! I totally missed this, fixed!